src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckPolicyMessageContentCollector.java

```diff
@@ -24,95 +24,41 @@
-public class BlackDuckPolicyMessageContentCollector extends MessageContentCollector {
+public abstract class BlackDuckPolicyMessageContentCollector extends MessageContentCollector {
     public static final String CATEGORY_TYPE = "policy";
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
-    @Autowired
-    public BlackDuckPolicyMessageContentCollector(final JsonExtractor jsonExtractor, final List<MessageContentProcessor> messageContentProcessorList) {
-        super(jsonExtractor, messageContentProcessorList, Arrays.asList(BlackDuckProviderContentTypes.POLICY_OVERRIDE, BlackDuckProviderContentTypes.RULE_VIOLATION, BlackDuckProviderContentTypes.RULE_VIOLATION_CLEARED));
+    public BlackDuckPolicyMessageContentCollector(final JsonExtractor jsonExtractor, final List<MessageContentProcessor> messageContentProcessorList, final Collection<ProviderContentType> contentTypes) {
+        super(jsonExtractor, messageContentProcessorList, contentTypes);
     }
 
-    @Override
-    protected void addCategoryItems(final List<CategoryItem> categoryItems, final JsonFieldAccessor jsonFieldAccessor, final List<JsonField<?>> notificationFields, final NotificationContent notificationContent) {
-        final List<JsonField<String>> categoryFields = getStringFields(notificationFields);
-        final List<LinkableItem> policyItems = getLinkableItemsByLabel(jsonFieldAccessor, categoryFields, BlackDuckProviderContentTypes.LABEL_POLICY_NAME);
-        final List<LinkableItem> componentItems = getLinkableItemsByLabel(jsonFieldAccessor, categoryFields, BlackDuckProviderContentTypes.LABEL_COMPONENT_NAME);
-        final List<LinkableItem> componentVersionItems = getLinkableItemsByLabel(jsonFieldAccessor, categoryFields, BlackDuckProviderContentTypes.LABEL_COMPONENT_VERSION_NAME);
-        final Optional<LinkableItem> firstName = getLinkableItemsByLabel(jsonFieldAccessor, categoryFields, BlackDuckProviderContentTypes.LABEL_POLICY_OVERRIDE_FIRST_NAME).stream().findFirst();
-        final Optional<LinkableItem> lastName = getLinkableItemsByLabel(jsonFieldAccessor, categoryFields, BlackDuckProviderContentTypes.LABEL_POLICY_OVERRIDE_LAST_NAME).stream().findFirst();
-        final SortedSet<LinkableItem> applicableItems = new TreeSet<>();
-        applicableItems.addAll(componentItems);
-        applicableItems.addAll(componentVersionItems);
+    protected void addApplicableItems(final List<CategoryItem> categoryItems, final Long notificationId, final LinkableItem policyItem, final String policyUrl, final ItemOperation operation, final SortedSet<LinkableItem> applicableItems) {
+        final List<String> keyItems = applicableItems.stream().map(LinkableItem::getValue).collect(Collectors.toList());
+        String[] keyArray = new String[keyItems.size() + 1];
+        keyArray = keyItems.toArray(keyArray);
+        keyArray[keyArray.length - 1] = policyUrl;
+        final CategoryKey categoryKey = CategoryKey.from(CATEGORY_TYPE, keyArray);
```

Comment:
Let's include a factory initializer that takes a List<String> rather than having to convert to an array here.